backend/api/src/create-announcement.ts

```diff
@@ -0,0 +1,57 @@
+import * as admin from 'firebase-admin'
+import { z } from 'zod'
+import { APIError, authEndpoint, validate } from './helpers'
+import { createCustomNotification } from 'shared/create-notification'
+import { getUser } from 'shared/utils'
+import { User } from 'common/user'
+import {
+  SupabaseDirectClient,
+  createSupabaseDirectClient,
+} from 'shared/supabase/init'
+import { isAdminId } from 'common/envs/constants'
+
+async function getAllUsersPaginated(pg: SupabaseDirectClient) {
+  const pageSize = 1000
+  let offset = 0
+  let users: User[] = []
+
+  while (true) {
+    const chunk = await pg.map(
+      `select data from users limit $1 offset $2`,
```

Comment:
I think it's good to have an `orderBy` in the query so that the offset searches over the same exact query pages.